addons/purchase_stock/models/purchase.py

```diff
@@ -56,6 +61,18 @@ def _compute_is_shipped(self):
             else:
                 order.is_shipped = False
 
+    @api.depends('picking_ids', 'picking_ids.state')
+    def _compute_receipt_status(self):
+        for order in self:
+            if order.state in ('done', 'cancel') or not order.picking_ids or all(p.state == 'cancel' for p in order.picking_ids):
```

Comment:
Does it still make sense to have PO state in the compute? It's not in the depends + it's invisible in the view before the pickings are made + the `receipt_status` isn't supposed to change when the PO state is changed to 'done' if I followed the chatter conversation correctly.